src/app/ReadClient.cpp

```diff
@@ -757,6 +800,24 @@ CHIP_ERROR ReadClient::ProcessAttributeReportIBs(TLV::TLVReader & aAttributeRepo
                 attributePath.mListOp = ConcreteDataAttributePath::ListOperation::ReplaceAll;
             }
 
+            if (attributePath ==
+                ConcreteDataAttributePath(kRootEndpointId, Clusters::IcdManagement::Id,
+                                          Clusters::IcdManagement::Attributes::OperatingMode::Id))
```

Comment:
please consider to construct OperatingMode in report, and inject into OnMessageReceive and validate this onPeerTypeChange in TestRead, thanks

Reply:
Test updated